NBitcoin/RPC/RPCClient.cs

```diff
@@ -2279,7 +2279,10 @@ public async Task<uint256[]> GenerateAsync(int nBlocks)
 					var result = (JArray)(await SendCommandAsync(RPCOperations.generate, nBlocks).ConfigureAwait(false)).Result;
 					return result.Select(r => new uint256(r.Value<string>())).ToArray();
 				}
-				catch (RPCException rpc) when (rpc.RPCCode == RPCErrorCode.RPC_METHOD_DEPRECATED || rpc.RPCCode == RPCErrorCode.RPC_METHOD_NOT_FOUND)
+				catch (RPCException rpc) when (
+					rpc.RPCCode == RPCErrorCode.RPC_METHOD_DEPRECATED 
+					|| rpc.RPCCode == RPCErrorCode.RPC_METHOD_NOT_FOUND 
+					|| (rpc.RPCCode == RPCErrorCode.RPC_MISC_ERROR && rpc.Message.StartsWith("generate\nhas been replaced by the -generate cli option")))

```

Comment:
please remove the check `StartsWith`, this is something that will break easily.

Reply:
Yes, it sucks really bad. I didn't find any other way to check for that specific error. I will remove it.